sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -31,10 +31,17 @@ import org.apache.spark.sql.catalyst.util.{CaseInsensitiveMap, CompressionCodecs
  * Most of these map directly to Jackson's internal options, specified in [[JsonParser.Feature]].
  */
 private[sql] class JSONOptions(
-    @transient private val parameters: CaseInsensitiveMap)
+    @transient private val parameters: CaseInsensitiveMap,
+    defaultColumnNameOfCorruptRecord: String)
   extends Logging with Serializable  {
 
-  def this(parameters: Map[String, String]) = this(new CaseInsensitiveMap(parameters))
+  def this(
+      parameters: Map[String, String],
+      defaultColumnNameOfCorruptRecord: String = "") = {
```

Comment:
Yes, it's really not a good solution, but it doesn't make sense to have a corrupt column name in all use cases. Picking another sentinel could inadvertently conflict with a real column. It should be `Option[String] = None` but this winds up being a large change that deserves a separate pull request.